app/scripts/controllers/network/util.js

```diff
@@ -1,24 +1,47 @@
 import { NETWORK_TO_NAME_MAP } from '../../../../shared/constants/network';
+import { TRANSACTION_ENVELOPE_TYPES } from '../../../../shared/constants/transaction';
 
 export const getNetworkDisplayName = (key) => NETWORK_TO_NAME_MAP[key];
 
 export function formatTxMetaForRpcResult(txMeta) {
-  return {
-    blockHash: txMeta.txReceipt ? txMeta.txReceipt.blockHash : null,
-    blockNumber: txMeta.txReceipt ? txMeta.txReceipt.blockNumber : null,
-    from: txMeta.txParams.from,
-    gas: txMeta.txParams.gas,
-    gasPrice: txMeta.txParams.gasPrice,
-    hash: txMeta.hash,
-    input: txMeta.txParams.data || '0x',
-    nonce: txMeta.txParams.nonce,
-    to: txMeta.txParams.to,
-    transactionIndex: txMeta.txReceipt
-      ? txMeta.txReceipt.transactionIndex
-      : null,
-    value: txMeta.txParams.value || '0x0',
-    v: txMeta.v,
-    r: txMeta.r,
-    s: txMeta.s,
+  const { r, s, v, hash, txReceipt, txParams } = txMeta;
+  const {
+    to,
+    data,
+    nonce,
+    gas,
+    from,
+    value,
+    gasPrice,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+  } = txParams;
+
+  const formattedTxMeta = {
+    v,
+    r,
+    s,
+    to,
+    gas,
+    from,
+    hash,
+    nonce,
+    input: data || '0x',
+    value: value || '0x0',
+    blockHash: txReceipt?.blockHash || null,
+    accessList: txReceipt?.accessList || null,
```

Comment:
This is tentative, as I'm not certain where in `txMeta` `accessList` lives

Reply:
txParams

Reply:
Thanks! fixed